Python/pywarpx/_libwarpx.py

```diff
@@ -234,17 +241,22 @@ def get_nattr():
     # --- The -3 is because the comps include the velocites
     return libwarpx.warpx_nComps() - 3
 
-def amrex_init(argv):
+def amrex_init(argv, mpi_comm=None):
     # --- Construct the ctype list of strings to pass in
     argc = len(argv)
     argvC = (_LP_c_char * (argc+1))()
     for i, arg in enumerate(argv):
         enc_arg = arg.encode('utf-8')
         argvC[i] = ctypes.create_string_buffer(enc_arg)
 
-    libwarpx.amrex_init(argc, argvC)
+    if mpi_comm is None:
+        libwarpx.amrex_init(argc, argvC)
+    else:
+        comm_ptr = MPI._addressof(mpi_comm)
+        comm_val = _MPI_Comm_type.from_address(comm_ptr)
+        libwarpx.amrex_init_with_inited_mpi(argc, argvC, comm_val)
```

Comment:
Runtime issue if build with `-DWarpX_MPI=OFF`:
> AttributeError: /home/runner/.local/lib/python3.8/site-packages/pywarpx/libwarpx.3d.so: undefined symbol: amrex_init_with_inited_mpi

We should guard this section so that we do not call missing symbols in serial builds.